### PR TITLE
Styled

### PR DESCRIPTION
**Pull request by Pomax**
- mofo-style'd yo
- jsbeautified
- errors flagged by jshint fixed
- all but a single jscs issue fixed (leaving https://github.com/mozilla/webmaker-app/issues/1425)

also do yourself a favour and look at https://github.com/mozilla/webmaker-app/pull/1426/files?w=0 with the whitespace diffs suppressed.

### CODE REVIEW

**Comment by thisandagain** [comment]
:smile: :dancer: :smile: :dancer: :smile: :dancer: :smile: :dancer: :smile: :dancer:

**Comment by gvn** [comment]
282-mother-of-god

File: gulp/jsbeautify.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+var beautifier = jsbeautify({
+  config: "./node_modules/mofo-style/linters/.jsbeautifyrc"
+});
+
+module.exports = function () {
+    return gulp.src(jspaths, {base: './'})
+      .pipe(beautifier)
+      .pipe(gulp.dest('./'));
+};
```
Comment (simonwex) [inline gulp/jsbeautify.js]: Need a `\n` here.

File: gulp/jsbeautify.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+var gulp = require('gulp');
+var jspaths = require('./util/jspaths');
+var jsbeautify = require('gulp-jsbeautifier');
+
+var beautifier = jsbeautify({
+  config: "./node_modules/mofo-style/linters/.jsbeautifyrc"
```
Comment (k88hudson) [inline gulp/jsbeautify.js]: single quotes

File: gulp/jscs.js [hunk 3]
```diff
@@ -1,7 +1,11 @@
 var gulp = require('gulp');
+var jspaths = require('./util/jspaths');
 var jscs = require('gulp-jscs');
 
+var fs = require("fs");
```
Comment (k88hudson) [inline gulp/jscs.js]: single quotes

File: gulp/jshint.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
 var gulp = require('gulp');
+var jspaths = require('./util/jspaths');
 var jshint = require('gulp-jshint');
 
 module.exports = function () {
-    return gulp.src('./{lib,blocks,components,views}/**/*.js')
-        .pipe(jshint())
+    return gulp.src(jspaths)
+        .pipe(jshint({ lookup: "./node_modules/mofo-style/linters/.jshintrc" }))
```
Comment (k88hudson) [inline gulp/jshint.js]: sinlge quotes

File: gulp/jscs.js [hunk 5]
```diff
@@ -1,7 +1,11 @@
...
+var config = fs.readFileSync("./node_modules/mofo-style/linters/.jscsrc");
```
Comment (k88hudson) [inline gulp/jscs.js]: personally I would specify `encoding` for `fs.readFIleSync` so it returns a string, not a buffer.

**Comment by thisandagain** [comment]
@Pomax could you get your build errors resolved and exclude #1425 so we can land this? I'd like to merge this quickly.

**Comment by k88hudson** [comment]
Yes let's do this ASAP to prevent more conflicts

File: gulp/jsbeautify.js (hunk 2 above)
Comment (Pomax) [inline gulp/jsbeautify.js]: I guess we want to beautify/jshint/jscs our gulp tasks, too.

File: gulp/jscs.js (hunk 5 above)
Comment (Pomax) [inline gulp/jscs.js]: is this a thing that we're okay with being a manual PR realisation, or do we want to somehow automated it as a "this is not wrong, but did you really want a buffer, or did you want a string?"

File: gulp/jsbeautify.js (hunk 2 above)
Comment (k88hudson) [inline gulp/jsbeautify.js]: yeah, that's a good idea

**Comment by Pomax** [comment]
I can do you one better. I added the gulp tasks to the beautify/lint/jscs task too because they're JS, why were they not in there, and did a minor tweak to the `App` constructor so that it doesn't need both `self` and `that` for closure escaping. I verified that with that change all the unit tests for storage still pass, so: :boom:

**Comment by Pomax** [comment]
So here's an interesting problem: our gulp tasks run a million times instead of once, when we use `gulp dev`, so if I add `jsbeautify` to the "clean" task, it'll run twice, where the second run happens while the first run is still happening, and the task tries to read files that are being written to, and everything breaks.

**Comment by Pomax** [comment]
:boom:

**Comment by k88hudson** [comment]
@Pomax what's the status on this one

**Comment by Pomax** [comment]
it needs review. And something landed that invalidated the PR, so it also needs an update.

File: gulpfile.js [hunk 6]
```diff
@@ -60,13 +62,11 @@ gulp.task('watch', function (done) {
     sequence('build-static', ['watchify', 'watchify-publish', 'watch-less'], 'cache', done);
 });
 
-// Test
 gulp.task('jshint', jshint);
 gulp.task('jscs', jscs);
 gulp.task('lint', ['jshint', 'jscs']);
 gulp.task('unit', ['build'], unit);
 gulp.task('test', ['lint', 'unit']);
-gulp.task('svgmin', svgmin);
```
Comment (gvn) [inline gulpfile.js]: Why is this being removed?

File: package.json [hunk 7]
```diff
@@ -37,9 +37,10 @@
     "gulp": "^3.8.8",
     "gulp-autoprefixer": "^2.0.0",
     "gulp-exit": "0.0.2",
+    "gulp-jsbeautifier": "0.0.4",
     "gulp-jscs": "1.3.1",
     "gulp-jshint": "1.8.4",
-    "gulp-less": "1.3.6",
+    "gulp-less": "^3.0.1",
```
Comment (gvn) [inline package.json]: Suggest locking this to 3.0.1

**Comment by gvn** [comment]
A couple comments. Generally seems ok to me. 

The refactoring of self = this's into binds feels a bit out of scope, but I'll allow it. :wink: 

If you can rebase and the tests pass then I'd say we're good. :+1:

File: gulpfile.js (hunk 6 above)
Comment (Pomax) [inline gulpfile.js]: that is an excellent question, I don't remember removing that so chucking it up to "accidental cmd-x" and will make sure that goes back in.

**Comment by Pomax** [comment]
yeah the bind(this) part isn't my favourite either, but it was a guaranteed way to make that `App` object portable to outside of where it lives right now, as prep for #1425 while also passing jscs. I'll be moving it out and undoing the bind in that ticket again =)

**Pull request merged**